code/modules/spells/roguetown/acolyte/general.dm

```diff
@@ -1,3 +1,87 @@
+//temple smite
+
+/obj/effect/proc_holder/spell/invoked/temple_smite
+	name = "Temple Smite"
+	desc = "Call divine punishment within your own church. Stuns a worthy target for 3 seconds and weakens them briefly. The gods will not punish clergy who are not excommunicated."
+	overlay_state = "sheltersmite"
+	releasedrain = 30
+	chargedrain = 0
+	chargetime = 0
+	range = 4
+	warnie = "sydwarning"
+	movement_interrupt = FALSE
+	sound = 'sound/magic/heal.ogg'
+	invocation_type = "none"
+	associated_skill = /datum/skill/magic/holy
+	antimagic_allowed = TRUE
+	recharge_time = 10 SECONDS
+	miracle = TRUE
+	devotion_cost = 10
+
+/obj/effect/proc_holder/spell/invoked/temple_smite/proc/_is_church_area(var/area/A)
+	if(!istype(A))
+		return FALSE
+	if(A:church_area)
```

Comment:
use `A.church_area` instead, : overrides the compile time check that the variable actually exists. 
see https://ref.harry.live/operator/colon vs https://ref.harry.live/operator/dot 

the istype checks should also be non-required if you're using a variable to check if an area is in the church, since it should be TRUE on those areas, and is therefore redundant-- 

Adding a variable to the base /area type also sucks if not necessary but isn't the end of the world.-- better solution might be a #define CHURCH_AREAS with a list of desired areas to be effected in; could then check `if(get_area(mob) in CHURCH_AREAS)`.